src/main/java/org/sagebionetworks/bridge/reporter/helper/BridgeHelper.java

```diff
@@ -51,11 +55,22 @@ public final void setBridgeCredentials(SignIn bridgeCredentials) {
 
     /*
      * Helper method to get all uploads for specified study and date range
+     * Paginated results should be added in one list altogether
      */
     public List<Upload> getUploadsForStudy(String studyId, DateTime startDateTime, DateTime endDateTime)
             throws IOException {
-        return sessionHelper(() -> bridgeClientManager.getClient(ForWorkersApi.class).getUploadsInStudy(studyId,
-                startDateTime, endDateTime).execute().body().getItems());
+
+        List<Upload> retList = new ArrayList<>();
+        String offsetKey = null;
+
+        do {
+            UploadList retBody = bridgeClientManager.getClient(ForWorkersApi.class).getUploadsInStudy(studyId,
+                    startDateTime, endDateTime, MAX_PAGE_SIZE, offsetKey).execute().body();
+            retList.addAll(retBody.getItems());
+            offsetKey = retBody.getOffsetKey();
+        } while (offsetKey != null);
```

Comment:
We need to sleep between each call. Calling Bridge as fast as possible defeats the purpse of pagination. Sleeping 1 second with each iteration makes the math easy, since that means we only need MAX_PAGE_SIZE per second ReadCapacity in DDB.